velox/exec/LocalPlanner.cpp

```diff
@@ -321,6 +324,17 @@ std::shared_ptr<Driver> DriverFactory::createDriver(
                 planNode)) {
       operators.push_back(
           std::make_unique<EnforceSingleRow>(id, ctx.get(), enforceSingleRow));
+    } else if (
+        auto assignUniqueId =
```

Comment:
Make sure to update maxDrivers's function in velox/exec/LocalPlanner.cpp to ensure that pipelines with AssignUniqueId nodes run single-threaded.

Reply:
@mbasmanova Curious why we need to limit this operator to be single-threaded? I use `atomic_int64_t` to generate the rowId so it should be able to be executed by multiple threads without id collison

Reply:
You are correct. I guess we can allow this operator to run multi-threaded as long as we use this atomic counter. In this case, there should be a counter per plan node ID as different AssignUniqueId nodes should not share a counter.